proposals/0012-introduce-reactNativeMetadata.md

```diff
@@ -0,0 +1,343 @@
+
+It's outside the scope of this RFC to declare all the new sections we intend to add, and we defer to the [Future proof and extensibility](#future-proof-and-extensibility) section for the process on how to include a new key to the `reactNativeManifest` section.
+
+## Excluded use cases
+
+This section lists the use cases where we believe the `reactNativeManifest` section should **not** be used.
+
+### React Native version support
+
+For the time being, we're not planning to add a `version` section to the `reactNativeManifest` section for libraries. The rationale is that we believe that the `react-native` version should be specified in either the `peerDependencies` or the `engines` section of the `package.json` of the project. This follows the idiomatic way to specify the version of a dependency in a NodeJS project.
+
+### TurboModule/Fabric toggles
+
+At the time of writing, we prefer to don't offer a dedicated section to toggle Fabric/TurboModule capability inside `reactNativeManifest`. The rationale is that we believe that the `newArch` section is sufficient to expose the New Architecture to users.
+Selectively toggling Fabric/TurboModule is a more advanced feature. We believe we'll still be offering a more advanced way to feature toggle such infrastructure pieces, but at the current state, our preference is to not expose such capability in the top level `reactNativeManifest` section.
+
+## Proposed Tooling
+
+This section lists the tooling we believe could be built on top of `reactNativeManifest` to improve the developer experience.
+
+### Build Logic support
+
+Build tools such as Gradle/CocoaPods or others should account for the `reactNativeManifest` section and use it to configure the build pipeline accordingly to this and future RFCs. Specifically they should follow this logic for a generic capability `foo`.
+
+1. If the capability `foo` is specificed in the `reactNativeManifest.capabilities` section, use the value specified in the `reactNativeManifest.capabilities.foo` section.
+   1. If the user is **also** specifying the capability `foo` in the build tool specific configuration (e.g. inside `gradle.properties` for Android) behave as follows:
+      1. If the two values are **compatible**, use them without notifying the user.
+      1. If the two values are **incompatible**, notify the user and use the value specified in the `reactNativeManifest.capabilities.foo` section will prevail and the value specified in the build tool specific configuration will be ignored.
+   1. If the user is **not** specifying the capability `foo` in the build tool specific configuration, honor the value specified in the `reactNativeManifest.capabilities.foo` section.
+1. If the capability `foo` is **not** specificed in the `reactNativeManifest.capabilities` section, honor the value specified in the build tool specific configuration or the default value if not specified.
```

Comment:
Can we be a little bit more aggressive?
So, for example, we can remove all the configuration flags in Cocoapods starting from version 0.73 (for example) so that the `reactNativeManifest` is the ONLY way to configure the app.
If there is no such file, we fail the build early, instructing the user how to set up those fields.

I know, it is a breaking change, but it will make the code much easier to write and follow and will let us converge to the ideal state faster. If the error messages are clear and actionable, most of the users will self-serve themselves.